app/views/employees/_form.html.erb

```diff
@@ -1,5 +1,11 @@
 <%= simple_form_for @employee do |form| %>
-  <%= form.input :slack_username %>
+  <%= form.input :slack_username,
+                 required: true,
+                 pattern: '^[^@]\w+[^ ]$',
+                 input_html: {
+                   title: I18n.t('employees.errors.slack_username')
```

Comment:
Yes now that I revisit the logic, I think you are onto something.

Rather than checking for whether a Slack username exists in our org in the controller via `unknown_employee(@employee.slack_username)`, we should add a validation in the `Employee` model that ensures the Slack username is included in the list of all slack usernames for the current org (http://guides.rubyonrails.org/active_record_validations.html#inclusion)

We could also add a uniqueness validation to `slack_username` so we wouldn't have to worry about `flash.now[:error] = "There is already a slack user with the username \"#{@employee.slack_username}\" in your organization."`

The way it is written currently *does* work but if we just did one check that would be sooo much better. something like:

```ruby
if @employee.save
  flash[:notice] = "Thanks for adding #{@employee.slack_username}"
  redirect_to root_path
else
   flash.now[:error] = @employees.errors.messages.to_sentence
      render action: :new
end
```

what do you think?